target/linux/realtek/files-6.6/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -2571,11 +2575,6 @@ static int __init rtl838x_eth_probe(struct platform_device *pdev)
 
 	/* Obtain device IRQ number */
 	dev->irq = platform_get_irq(pdev, 0);
-	if (dev->irq < 0) {
-		dev_err(&pdev->dev, "cannot obtain network-device IRQ\n");
-		return err;
-	}
-
```

Comment:
why do you remove this?

Reply:
The Coccinelle check says to remove the dev_err call since it gets called by platform_get_irq. I figure request_irq can handle returning on error, which it will.

Reply:
It prints an error mesage, but you should still return an error to the caller, see the example here: https://elixir.bootlin.com/linux/v6.11.8/source/drivers/base/platform.c#L243

Reply:
Yeah and https://elixir.bootlin.com/linux/v5.5.2/source/kernel/irq/manage.c#L1932 has multiple returns, which prints an error right after platform_get_irq.